### PR TITLE
Improve mobile UX

### LINKED ISSUE #93: Move search field to the website header
Search should be available from anywhere in the website

### LINKED ISSUE #104: Improve UX on mobile
## Updated issue

- Issue: Update the design for mobile to improve UX
- Corresponding MR: #105

## Original RFC for the design study

While surfing on the mobile site of Lutris I noticed several minor issues, for example:

- Lutris brand breaking on small screens
- A lot of wasted space (both horizontal and vertical)
- A lot of clutter (unneeded information, ...)

So I decided to take half an hour and experiment a bit. Below are the results, and I would like to start a discussion on whether I should further pursue development leading to a merge request, with your feedback of course.

Note that the aim for a development in this direction would only be to tweak the mobile version of Lutris.net.

Another minor notes:

- Filter icon instead of a huge list of platforms and genres. The icon should expand a mobile-optimised filter menu.
- I did omit the search function, because there's another open issue discussing about moving the search to a global place.

design-study

### PR DESCRIPTION
**Pull request by charlydelta**
# TL;DR

- Fixes #93 
- Fixes #104 
- General purpose: Improve mobile UX by tweaking CSS/HTML
- **Some changes affect non-mobile views!** However, it's intended.
- Fix an issue where the Lutris logo would break the header on XS
- Remove wasted horizontal/vertical space
- Remove clutter
- Manifold of minor fixes which I didn't bother noting down

# Screenshots

**Please note**: All game banners are low quality because I was too lazy to properly setup the database.

<details>
<summary><strong>Figure 1</strong>: First proof-of-concept. Left: original; Right: improved version</summary>

design-study

</details>

<details>
<summary><strong>Figure 2</strong>: <code>/games</code> page. Top nav: global search and menu item. Header nav: submit game, filter</summary>

screen shot 2017-11-03 at 02 08 33

</details>

<details>
<summary><strong>Figure 3</strong>: Opened filter page</summary>

screen shot 2017-11-03 at 02 08 39

</details>

<details>
<summary><strong>Figure 4</strong>: Reworked pagination for page 1</summary>

screen shot 2017-11-03 at 02 08 46

</details>

<details>
<summary><strong>Figure 5</strong>: Reworked pagination for page >= 3</summary>

screen shot 2017-11-03 at 02 08 53

</details>

<details>
<summary><strong>Figure 6</strong>: <code>/library</code> page</summary>

screen shot 2017-11-03 at 02 09 04

</details>

<details>
<summary><strong>Figure 7</strong>: <code>/download</code> page</summary>

screen shot 2017-11-03 at 02 09 11

</details>

<details>
<summary><strong>Figure 8</strong>: <code>/</code> page with global search in action</summary>

screen shot 2017-11-03 at 02 09 35

</details>

<details>
<summary><strong>Figure 9</strong>: NON-MOBILE: <code>/</code> page. Global search available as well</summary>

screen shot 2017-11-03 at 02 09 53

</details>

<details>
<summary><strong>Figure 10</strong>: NON-MOBILE: <code>/</code> page. Global search in action</summary>

screen shot 2017-11-03 at 02 09 58

</details>

<details>
<summary><strong>Figure 11</strong>: NON-MOBILE: <code>/games</code> page. Reworked filter</summary>

screen shot 2017-11-03 at 02 10 09

</details>

<details>
<summary><strong>Figure 12</strong>: NON-MOBILE: <code>/library</code> page</summary>

screen shot 2017-11-03 at 02 10 15

</details>

<details>
<summary><strong>Figure 13</strong>: NON-MOBILE: New pagination here, too. Page 2</summary>

screen shot 2017-11-03 at 02 10 24

</details>

# Reasoning

- Header:
  - Just show brand on small screens, not the name
  - Show menu and global search for XS screens
  - Global search (for all screen-sizes) as per #93
- Pagination:
  - Old pagination showed way too much data which cluttered the view (must of the data was not relevant)
  - Old pagination links were really hard to click, even with the mouse. Nearly impossible on mobile to hit
  - New pagination for all screen sizes, since the above mentioned points also apply here
- Filter:
  - Genres and Platforms were a bad UX because you had to scroll endlessly to find the right thing. On mobile it was even harder. Tag-cloud like design is better suited in this case, even though I have to say that it still is not optimal
- Pages / general
  - Reworked most pages so they have a `<h1>` header (uniformity + conformity)

# ToDo's

- [x] Smaller paddings/margins in general for XS
- [x] Fix Lutris logo breaking the header on XS
- [x] Smaller headings on XS
- [x] Come up with a appropriate replacement until #93 is resolved, or just resolve #93 
- [x] Improve pagination for mobile use
- [x] Make a filter for mobile
  - [x] Don't display on default, only when needed (see proof-of-concept)
  - [x] Evaluate if existing template can be reused for this purpose
  - [x] Integrate button into the page where it makes sense. (Contrary to proof-of-concept screenshot above: The filter icon is placed in the breadcrumbs-`div`. This does not make sense at all. A new line, on the other hand, would waste horizontal and vertical space)
- [x] Relocate "Submit game" button
- [x] Rework "Download" page
- [x] Rework "Library" page
- [x] Unify layout

# Dependencies

- [x] #93: FIXED ~~It is very uncomfortable and counter-intuitve for the search to be *under* `/games`, below all games on that page, so it should be removed from there, but not without replacement though. So #93 must be done for this MR to be merged~~

# Future work

- Refactor `.less` handling: Don't use one global less file, but make logic units (one file per component for example, with a global file for general purpose)
- More re-usable components to make the look&feel more consistent

### CODE REVIEW

**Comment by charlydelta** [comment]
@strycore Travis fails with `TemplateDoesNotExist: docs/installers.html`. Does `make doc` need to be run before running `make test`?

**Comment by strycore** [comment]
@charlydelta I started setting up Travis but it's not fully completed yet, I guess `make docs` should fix the issue indeed.

**Comment by charlydelta** [comment]
@strycore Which is awesome, thank you for that! One thing that you also should consider: In my other MR I made changes to the database. Since another DB change could be commited in the meantime, it is possible that we end up with merge conflicts in the migration files, so I excluded them on purpose. Of course this will make all fails to fail. So you should probably call `python manage.py makemigrations` beforehand as well.

**Comment by charlydelta** [comment]
**Interim report**

Two screenshots from the current commit

Things to note:

- Reworked the `game_list` view (remove clutter + unneeded spaces)
- Reworked pagination (The old pagination is not very good from a UX perspective, because there was a lot of data displayed which nobody actually cares about, like the total number of entries in the database etc. This is a minimal approach with better clickable items and it was also integrated into the non-mobile view)
- Integrated filter-button into the top-nav
- Integrated a filter collapsable (currently only a copy, refactoring to be done after the design phase)

<details>
<summary>A screenshot of the /games page, top to bottom (list truncated) (low-quality game banners, but thats because of my dummy data images which are only 184x90 or something)</summary>

screen shot 2017-11-01 at 13 42 54

</details>

<details>
<summary>A screenshot of the mobile filter page</summary>

screen shot 2017-11-01 at 13 43 09

</details>

**Comment by charlydelta** [comment]
**Ready for review**

I think I am done here. I added a new section *Reasoning* to the OP where I tried to justify all my design decisions, however I am sure that I missed one or the other point which you could like to know more. Please feel free to ask me anything in that regard.

The current state, 2f57ce6, is represented by the screenshots **Figure 2** - **Figure 13** in the OP. **All other screenshots show out-of-date versions**

I am grateful for feedback.

**Comment by strycore** [comment]
This looks awesome, many thanks!
